samples/snippets/type_system_test.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def test_type_system_examples() -> None:
+    # [START bigquery_dataframes_type_sytem_local_type_conversion]
+    import pandas as pd
+
+    import bigframes.pandas as bpd
+
+    s = pd.Series([pd.Timestamp("20250101")])
+    print(s.dtype)
+    # datetime64[ns]
+    print(bpd.read_pandas(s).dtype)
+    # timestamp[us][pyarrow]
```

Comment:
Might be good to confirm that these comments are accurate with some assertions. Otherwise, we won't know when our docs are incorrect and we're possibly making breaking changes.

```suggestion
    pandas_dtype = s.dtype
    assert str(pandas_dtype) == "datetime64[ns]"
    bigframes_dtype = bpd.read_pandas(s).dtype
    assert str(bigframes_dtype) == "timestamp[us][pyarrow]"
```

Same applies to the other samples where we want to demonstrate bigframes behavior.

In the cases where an assertion in the code sample itself doesn't make sense, we can put the assertions after the `[END ...]` comment and before the next sample starts.

Reply:
Sure. Tests added.